data/transactions/logic/assembler.go

```diff
@@ -1077,9 +1115,9 @@ func asmDefault(ops *OpStream, spec *OpSpec, args []string) error {
 		switch imm.kind {
 		case immByte:
 			if imm.Group != nil {
-				fs, ok := imm.Group.SpecByName(args[i])
+				fs, ok := imm.Group.SpecByName(args[i].str)
 				if !ok {
-					_, err := byteImm(args[i], "")
+					_, err := byteImm(args[i].str, "")
 					if err == nil {
 						// User supplied a uint, so we see if any of the other immediates take uints
```

Comment:
this method baffles me.  Random non-opcode bytes fall into here?  "Other immediates take uints"?

Reply:
It's trying very hard to eplain what went wrong.  If it couldn't produce an `fs` on 1118, maybe it's because the user gave an int instead of a symbol. And maybe _that's_ because the screwed up argument order, like
`txna 2 Accounts` instead of `txna Accounts 2`
In that case, tell them the 2 is in the wrong spot.
And then it does something similar for symbols that might be in the wrong place.
I probably wouldn't have gone through the trouble to give some detailed errors. Ilan did it, and it seems a shame to remove the extra help.

Reply:
oh I see now, that's helpful thanks.